crates/runtime/benches/bench.rs

```diff
@@ -293,17 +312,30 @@ async fn record_explain_plan(
         .map_err(|e| format!("query `{query}` to results: {e}"))?;
 
     let Ok(explain_plan) = arrow::util::pretty::pretty_format_batches(&plan_results) else {
-        panic!("Failed to format plan");
+        return Err("Failed to format plan".to_string());
     };
+
+    let mut assertion_err: Option<String> = None;
+
     insta::with_settings!({
         description => format!("Query: {query}"),
         omit_expression => true,
         filters => vec![
             (r"required_guarantees=\[[^\]]*\]", "required_guarantees=[N]"),
         ],
     }, {
-        insta::assert_snapshot!(format!("{connector}_{query_name}_explain"), explain_plan);
+        let result = panic::catch_unwind(|| {
+            insta::assert_snapshot!(format!("{connector}_{query_name}_explain"), explain_plan);
+        });
+        if result.is_err() {
+            assertion_err = Some(format!("Snapshort assertion failed for {connector}, {query_name}"));
+        }
     });
+
+    if let Some(assertion_err) = assertion_err {
+        return Err(assertion_err);
+    }
```

Comment:
```suggestion
    assertion_err.ok_or_else(Err)?;
```

Does this work? (I haven't tested it)

Reply:
`ok_or_else` only works as follows
```
    assertion_err
        .clone()
        .ok_or_else(|| assertion_err.unwrap_or_default())?;
```